storage/storage-client/src/lib.rs

```diff
@@ -275,6 +277,51 @@ impl StorageRead for StorageReadServiceClient {
             .boxed();
         Ok(stream)
     }
+
+    // TODO migrate this to `ConfigStorage` trait as non-async
+    // and implement for LibraDBTrait once `StorageRead` is deprecated
+    async fn batch_fetch_config(&self, access_paths: Vec<AccessPath>) -> Vec<Option<Vec<u8>>> {
+        let mut requests = vec![];
+        // some access paths can have the same address, so don't duplicate requests for them
+
+        let mut req_addresses = vec![];
+        for path in access_paths.iter() {
+            let address = path.address;
+            if !req_addresses.contains(&address) {
+                req_addresses.push(address.clone());
+                requests.push(RequestItem::GetAccountState { address });
+            }
+        }
+        let responses = if let Ok(responses) = self.update_to_latest_ledger(0, requests).await {
```

Comment:
nit: match more idiomatic?

Reply:
but with match we have the extra indentation?